scheduling/tasks.go

```diff
@@ -70,50 +77,61 @@ func (t *TaskMetadata) Unmarshal(data []byte) error {
 
 	var temp tempConfig
 	if err := json.Unmarshal(data, &temp); err != nil {
-		return fmt.Errorf("failed to deserialize task metadata: %w", err)
+		errMessage := fmt.Errorf("failed to deserialize task metadata: %w", err)
+		return fferr.NewInternalError(errMessage)
 	}
 	if temp.ID == 0 {
-		return fmt.Errorf("task metadata is missing ID")
+		return fferr.NewInvalidArgumentError(fmt.Errorf("task metadata is missing ID"))
```

Comment:
nit: for readability we should do all the validations first (maybe even in a separate function) and then the rest